dcos_launch/platforms/aws.py

```diff
@@ -131,14 +131,16 @@ def create_key_pair(self, key_name):
     def get_service_resources(self, service, resource_name):
         """Return resources and boto wrapper in every region for the given boto3 service and resource type."""
         for region in aws_region_names:
-            for resource in getattr(self.resource(service, region['id']), resource_name).all():
-                yield resource
+            # line below is needed because function get_all_stacks needs to copy the boto wrapper with the correct
+            # region when initializing each CfStack object
+            self.region = region['id']
```

Comment:
Would anything else expect this to be changed back to the original value after the resources are gone though?

Reply:
@margaret  Nope! the only tools that use this function are cleanup tools that need access to all resources in all regions, so they never use a boto wrapper with a specific region. They don't use the region member at all. They are initialized with region=None.